src/test/java/org/testobject/appium/junit/AppiumDriverCalculatorSuiteTestJUnit.java

```diff
@@ -43,17 +44,18 @@ public void setup() throws MalformedURLException {
 	@Test
 	public void twoPlusTwoOperation() {
 
-		MobileElement buttonTwo = (MobileElement)(driver.findElement(By.id("net.ludeke.calculator:id/digit2")));
-		MobileElement buttonPlus = (MobileElement)(driver.findElement(By.id("net.ludeke.calculator:id/plus")));
-		MobileElement buttonEquals = (MobileElement)(driver.findElement(By.id("net.ludeke.calculator:id/equal")));
-		MobileElement resultField = (MobileElement)(driver.findElement(By.xpath("//android.widget.EditText[1]")));
+		MobileElement buttonTwo = (MobileElement) (driver.findElement(By.id("net.ludeke.calculator:id/digit2")));
+		MobileElement buttonPlus = (MobileElement) (driver.findElement(By.id("net.ludeke.calculator:id/plus")));
+		MobileElement buttonEquals = (MobileElement) (driver.findElement(By.id("net.ludeke.calculator:id/equal")));
+		By resultFieldBy = By.xpath("//android.widget.EditText[1]");
 
 		buttonTwo.click();
 		buttonPlus.click();
 		buttonTwo.click();
 		buttonEquals.click();
 
-		(new WebDriverWait(driver, 30)).until(ExpectedConditions.textToBePresentInElement(resultField, "4"));
+		WebDriverWait wait = new WebDriverWait(driver, 30);
+		assertEquals(wait.until(d -> d.findElement(resultFieldBy).getText().trim()), "4");
```

Comment:
this is quite a long statement to make in an assert. maybe we can pull the wait.until method out into a separate variable